### PR TITLE
refactor(http_sender): simplify

### PR DESCRIPTION
**Pull request by tamird**
Discovered while working on retry stuff.

### CODE REVIEW

**Comment by MBerhault** [comment]
this seems excessive. "GET" and "POST" are much more readable than util.GetMethod. Even the standard library doesn't const those out.

**Comment by tamird** [comment]
> Even the standard library doesn't const those out.

The standard library only consts out status codes =/

File: server/admin.go [hunk 1]
```diff
@@ -144,9 +144,9 @@ func (s *adminServer) handleZoneAction(w http.ResponseWriter, r *http.Request) {
 // handleRESTAction handles RESTful admin actions.
 func (s *adminServer) handleRESTAction(handler actionHandler, w http.ResponseWriter, r *http.Request, prefix string) {
 	switch r.Method {
-	case "GET":
+	case util.GetMethod:
 		s.handleGetAction(handler, w, r, prefix)
-	case "PUT", "POST":
+	case "PUT", util.PostMethod:
```
Comment (bdarnell) [inline server/admin.go]: Might as well make constants for PUT and DELETE too.

File: client/http_sender_test.go [hunk 2]
```diff
@@ -63,7 +63,7 @@ func startTestHTTPServer(handler http.Handler) (*httptest.Server, string) {
 // TestHTTPSenderSend verifies sending posts.
 func TestHTTPSenderSend(t *testing.T) {
 	server, addr := startTestHTTPServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		if r.Method != "POST" {
+		if r.Method != util.PostMethod {
```
Comment (petermattis) [inline client/http_sender_test.go]: I'd be more excited about this type of change if it was actually fixing a bug. As it is, this is less readable than simply typing out `"POST"`.

**Comment by tamird** [comment]
The people have spoken. Removing the first commit.

**Comment by bdarnell** [comment]
LGTM

**Comment by petermattis** [comment]
LGTM for the `http_sender.go` change.

**Comment by tamird** [comment]
Removed `post` entirely. PTAL if you want, the change is minor.

File: client/http_sender.go [hunk 3]
```diff
@@ -99,90 +92,60 @@ func (s *httpSender) Send(_ context.Context, call proto.Call) {
...
-// are protobuf-serialized and written as the POST body. The content
-// type is set to application/x-protobuf.
-//
-// On success, the response body is unmarshalled into call.Reply.
-func (s *httpSender) post(call proto.Call) (*http.Response, error) {
 	// Marshal the args into a request body.
 	body, err := gogoproto.Marshal(call.Args)
 	if err != nil {
-		return nil, err
+		return err
```
Comment (petermattis) [inline client/http_sender.go]: `Send` doesn't return a value. How is this compiling? Or am I misreading the code?

**Comment by MBerhault** [comment]
LGTM

File: client/http_sender.go (hunk 3 above)
Comment (MBerhault) [inline client/http_sender.go]: it isn't: from circleci:
# github.com/cockroachdb/cockroach/client
client/http_sender.go:98: too many arguments to return

Comment (tamird) [inline client/http_sender.go]: oops

Comment (tamird) [inline client/http_sender.go]: this is why i kept `post` originally. reverting.

**Pull request merged**